Packs/HackerOne/Integrations/HackerOne/HackerOne.py

```diff
@@ -0,0 +1,677 @@
+       - 'limit' argument cannot be specified with 'page' or 'page_size' argument.
+       - 'page_size' argument is within its expected lower/upper bounds.
+       - If 'limit' is not given, and 'page_size' is, sets 'limit' value to 'page_size' value.
+       - If 'limit' is not given, and 'page_size' is not given, sets 'limit' value to 'DEFAULT_PAGE_SIZE' value.
+       Args:
+           args (Dict): Demisto argument.
+       Returns:
+           - (int, int): 'page', 'limit' extracted, or their default values used.
+           - (DemistoException): If arguments don't follow the expected logic mentioned.
+       """
+    page = get_and_validate_positive_int_argument(args, 'page_number', lower_bound=LOWER_BOUND, upper_bound=INT32)
+    page_size = get_and_validate_positive_int_argument(args, 'page_size', lower_bound=LOWER_BOUND,
+                                                       upper_bound=MAXIMUM_PAGE_SIZE)
+    limit = get_and_validate_positive_int_argument(args, 'limit', lower_bound=LOWER_BOUND, upper_bound=MAXIMUM_LIMIT)
+    if limit and (page_size or page):
+        raise ValueError(MESSAGES["INVALID_ARGUMENT"])
+    if not limit and page_size:
+        limit = page_size
+    if not limit:
+        limit = DEFAULT_PAGE_SIZE
+    if not page:
+        page = LOWER_BOUND
+
+    return page, limit
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises:
+     exceptions if something goes wrong.
+
+    Args:
+        client: client to use
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+
+    if demisto.params().get('isFetch'):
+        fetch_incidents(client, {})
+    else:
+        client.program_list(params={"page[size]": 1})
+
+    return 'ok'
+
+
+def fetch_incidents(client: Client, last_run: dict, ) -> Tuple[dict, list]:
+    """Fetches incidents from HackerOne.
+
+    :type client: ``Client``
+    :param client: client to use
+
+    :type last_run: ``Dict[str, str]``
+    :param last_run: Last run returned by function demisto.getLastRun
+
+    :type params: ``Dict[str, str]``
+    :param params: Arguments for fetch-incident.
+
+    :rtype: ``Tuple``
+    :return: tuple of dictionary of next run and list of fetched incidents
+    """
+    validate_fetch_incidents_parameters(client.max_fetch, client.program_handle, client.filters)
+    fetch_params = prepare_fetch_incidents_parameters(client.max_fetch, client.first_fetch, client.program_handle,
+                                                      client.severity, client.state, client.filters)
+    first_fetch = fetch_params["filter[created_at__gt]"]
+
+    fetch_params["page[number]"] = last_run.get("next_page", 1)
+    fetch_params["filter[created_at__gt]"] = last_run.get("next_created_at", first_fetch)  # type: ignore
```

Comment:
This is more efficient in terms of setting the fetch time. Also, let's set the page in the prepare_fetch_incidents_parameters function as well.
```suggestion
    validate_fetch_incidents_parameters(client.max_fetch, client.program_handle, client.filters)
    time_to_fetch = last_run.get("next_created_at", first_fetch)  # type: ignore
    page_to_fetch = last_run.get("next_page", 1)
    fetch_params = prepare_fetch_incidents_parameters(client.max_fetch, client.program_handle,
                                                      client.severity, client.state, time_to_fetch, page_to_fetch, client.filters)    
```